bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -531,6 +574,61 @@ private void initEncodingTypes(CaseDbConnection connection) throws SQLException,
 			releaseSingleUserCaseWriteLock();
 		}
 	}
+	
+	/**
+	 * Set up or update the hasChildren map using the tsk_objects table.
+	 * 
+	 * @param connection
+	 * @throws TskCoreException 
+	 */
+	private void populateHasChildrenMap(CaseDbConnection connection) throws TskCoreException {
+		long timestamp = System.currentTimeMillis();
+		
+		
+		Statement statement = null;
+		ResultSet resultSet = null;
+		acquireSingleUserCaseWriteLock();
+		try {
+			statement = connection.createStatement();
+			resultSet = statement.executeQuery("select distinct par_obj_id from tsk_objects"); //NON-NLS
+			
+			synchronized (this) {
+				if (hasChildrenBitSetMap == null) {
+					hasChildrenBitSetMap = new HashMap<Long, SparseBitSet>();
+				}
+				while(resultSet.next()) {
+					setHasChildren(resultSet.getLong("par_obj_id"));
+				}
+			}
+			long delay = System.currentTimeMillis() - timestamp;
+			logger.log(Level.INFO, "Time to initialize parent node cache: {0} ms", delay); //NON-NLS
```

Comment:
Should we be retaining this performance data for the release, or is this more along the lines of debug info that can now be removed?

Reply:
I figured leave it in at least for this release so we can make sure this isn't causing a massive slowdown for opening cases.

Reply:
Okay.